server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -461,29 +467,35 @@ protected void doRun() {
                             MappingMetadata mappingMd = indexMetadata.mapping();
                             Version indexCreated = indexMetadata.getCreationVersion();
                             indexRequest.resolveRouting(metadata);
+                            boolean idIsAutoGenerated = indexRequest.id() == null;
```

Comment:
Yeah - I was looking at that but didn't do it because I didn't like that there were two ways to know depending on where you are in the process. If you like the method I can add it though. It makes sense. Its just funny there are two ways to tell.